sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -508,7 +501,7 @@ private Status getStatusWithDefault() {
   }
 
   SpanId getParentSpanId() {
-    return parentSpanId;
+    return TracingContextUtils.getSpan(parent).getContext().getSpanId();
```

Comment:
I don't quite understand what we're getting from the change in this PR. It seems we round-trip through `Context` API instead of much more efficiently accessing fields directly?

Reply:
This spec change (this PR is an implementation of) has two purposes:

1. Enable passing arbitrary information between propagators inject, extract and exporters (and in the future, after #856 and #881 the samplers)
2. Encourage users to properly propagate the context (it's easier to use Tracer.withSpan than doing the whole dance of `TracingContextUtils.contextWithSpan(parentSpan, Context.current()` which is intentionally made more cumbersome. For example, see the [Actor propagation change](https://github.com/open-telemetry/opentelemetry-java/pull/1611/files#diff-fb2061671d30874e4251c2ff1f7c41baR48) which previously contained what I would call a bug, as it lost all information in the Context except for the span.

Reply:
This no longer applies.